Support for moving average in elasticsearch data source.

bergquist opened the pull request:
Some basic functionality for moving average in the ES data source.

Only support default options so far.

Commit:
added alerting tab stub and styles

Commit:
added global alerts list stub and styles

Commit:
updated-list-views: added filter-table less component, updating styles for data sources table

Commit:
updated-list-views: added filter controls, updated tables on API Keys and Data Sources

Commit:
updated-list-views: updating table layout for org users

Commit:
updated-list-views: added new table layout to additional pages

mtanda committed:
update aws-sdk-go v1.0.0

mtanda committed:
(cloudwatch) use paging

mtanda committed:
add separate datasource parameter withCredentials

Commit:
Merge branch 'master' of github.com:grafana/grafana

Commit:
removed filters and actions for now

Commit:
Merge branch 'master' of github.com:grafana/grafana

Commit:
merge with master, conflicts fixed

Commit:
removed redundant labels

Commit:
Merge branch 'master' of github.com:grafana/grafana

Commit:
Merge branch 'master' into updated-list-views

torkelo committed:
Merge branch 'cloudwatch_aws-sdk-go-v1.0.0' of https://github.com/mtanda/grafana into mtanda-cloudwatch_aws-sdk-go-v1.0.0

mtanda committed:
add with_credentials to datasource model

torkelo committed:
fix(timepicker): changed name of Day so far -> Today so far, Week to date > Week so far, closes #3465

torkelo committed:
Merge branch 'updated-list-views' of https://github.com/nchristus/grafana into nchristus-updated-list-views

mtanda committed:
add withCredentials checkbox

torkelo committed:
refactor(table views): #3466

torkelo committed:
Merge branch 'with_credentials' of https://github.com/mtanda/grafana into mtanda-with_credentials

torkelo committed:
refactor(http data source options): #3473

torkelo committed:
fix(panel): removed accidentally commited test markup

torkelo committed:
fix(graph): removed experimental mockup code that was accidently merged master

torkelo committed:
fix(timerange): fixed broken unit tests

torkelo committed:
Merge branch 'moving_avg_es_support' of https://github.com/bergquist/grafana into bergquist-moving_avg_es_support

torkelo commented on public/app/plugins/datasource/elasticsearch/metric_agg.js:
```diff
@@ -13,14 +13,21 @@ function (angular, _, queryDef) {
+    };
+
+    $scope.updateMavgOptions = function() {
+      $scope.mavgOptions = queryDef.getMovingAverageOptions($scope.target);
```
this should be named something more generic since I think we can reuse this for all pipeline aggregations (like derivative)

torkelo committed:
Merge branch 'moving_avg_es_support' of https://github.com/bergquist/grafana into bergquist-moving_avg_es_support

torkelo committed:
feat(elasticsearch): added pipleline aggregation derivative

torkelo committed:
feat(elasticsearch): metric options alignement

bergquist committed:
show pipeline agg source for derivative

bergquist committed:
feat(elasticsearch): add pipeline settings

bergquist committed:
feat(elasticsearch): update pipeline aggs if type change

make it possible to switch between different pipeline
aggregates without causing problms in the ui

bergquist committed:
feat(elasticsearch): improve pipeline aggs structure

bergquist committed:
refactor(elasticsearch): mavg naming -> pipeline agg

bergquist committed:
feat(elasticsearch): remove pipeline aggs as possible sources

bergquist committed:
feat(elasticsearch): make series naming generic for pipeline aggs

torkelo commented:
looking good :)

bergquist committed:
feat(elasticsearch): adds last class for derivatives

bergquist committed:
feat(elasticsearch): add support for model setting

bergquist committed:
feat(elasticsearch): display more info in options
